tests/test_main.py

```diff
@@ -365,3 +365,19 @@ def test_not_render_rules_also_matched(z9324d_extractor, capsys):
     assert "act as TCP client" in std.out
     assert "connect TCP socket" in std.out
     assert "create TCP socket" in std.out
+
+
+# It tests main works with different backends. It doesn't test that the backend
```

Comment:
It doesn't test that the backend is actually called. We could use something like [mock](https://pypi.org/project/mock) to test really test it, but I think that adding a dependency just for this is not worthwhile. We could also print which backend has been used, as this is really easy to test. But I would only do this if we think it is a useful information for the user. I think it can make sense in the `-vv` mode.

Reply:
In verbose and vverbose mode the output contains the extractor, e.g. `VivisectFeatureExtractor` and `std.out` could be tested for it.